ios/Exponent/Kernel/DevSupport/EXKernelDevKeyCommands.m

```diff
@@ -68,6 +68,46 @@ - (NSString *)description
 @interface EXKernelDevKeyCommands ()
 
 @property (nonatomic, strong) NSMutableSet<EXKeyCommand *> *commands;
++ (void)handleKeyboardEvent:(UIEvent *)event;
+
+@end
+
+@interface UIEvent (UIPhysicalKeyboardEvent)
+
+@property (nonatomic) NSString *_modifiedInput;
+@property (nonatomic) NSString *_unmodifiedInput;
+@property (nonatomic) UIKeyModifierFlags _modifierFlags;
+@property (nonatomic) BOOL _isKeyDown;
+@property (nonatomic) long _keyCode;
+
+@end
+
+
+@implementation UIApplication (EXKeyCommands)
+
+- (void)EX_handleKeyUIEventSwizzle:(UIEvent *)event
+{
+  BOOL interactionEnabled = !UIApplication.sharedApplication.isIgnoringInteractionEvents;
+  BOOL hasFirstResponder = NO;
+  
+  if (interactionEnabled) {
+    UIResponder *firstResponder = nil;
+    for (UIWindow *window in [self windows]) {
+      firstResponder = [window valueForKey:@"firstResponder"];
+      if (firstResponder) {
+        hasFirstResponder = YES;
+        break;
+      }
+    }
+    
+    if (firstResponder && [firstResponder isKindOfClass: [UITextField class]]) {
```

Comment:
react-native's `TextInput` can also be `UITextView`

https://github.com/facebook/react-native/blob/9187e207524fd1a3e916a0dedb4a5dbdb7fbe89e/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm#L38-L39